setup.py

```diff
@@ -1,30 +1,63 @@
-import glob
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, immuneML Development Team.
+# Distributed under the LGPLv3 License. See LICENSE for more info.
+"""immuneML.
+
+This file is part of immneML.
 
+immuneML is free software, please our term and conditions detailed
+in the file LIVENCE.md
+"""
+import ast
+import glob
+import pathlib
+from codecs import open as openc
 from setuptools import setup, find_packages
 
-from source.environment.Constants import Constants
+FULL_VERSION = '0.9.0.dev0'  # Automatically set by setup_version.py
+
+
+def get_long_description():
+    """Return the contents of README.md as a string."""
+    here = pathlib.Path(__file__).absolute().parent
+    long_description = ''
+    with openc(here.joinpath('README.md'), encoding='utf-8') as fileh:
+        long_description = fileh.read()
+    return long_description
+
+
+def get_version():
+    """Return the version from version.py as a string."""
+    here = pathlib.Path(__file__).absolute().parent
+    filename = here.joinpath('source', 'version.py')
+    with openc(filename, encoding='utf-8') as fileh:
+        for lines in fileh:
+            if lines.startswith('FULL_VERSION ='):
+                version = ast.literal_eval(lines.split('=')[1].strip())
+                return version
+    return FULL_VERSION
 
 
 def import_requirements(filename) -> list:
+    """Import requirements."""
     with open(filename, 'r') as file:
         requirements = file.read().split("\n")
     return requirements
 
 
 setup(
-    name="immune-ml",
-    version=Constants.VERSION,
+    name="immuneML",
+    version=get_version(),
```

Comment:
we already had a version numbering system which we have used for a few months, will that be overwritten/start over?

Reply:
I uses a pipy package installation number/labeling. 

The name is important to have immuneML (the pipy packages follows that sintax)
If you like this version labeling, yes, it would start over. The important thing is to have a good system once the package will be public in order to keep track of the releases and be able to reproduce the bug reports.

Reply:
So the issue is that pipy uses a different version formatting?
But if we use this differently-formatted versioning, we might still want to start from another starting point version that more closely resembles the current version number?
(maybe see what Milena thinks about that)

Reply:
Because i think we may reference to the used version numbers in the paper, and we reference to them in the documentation. So it is strange if in the docs it talks about version 1.0.1, and we start counting below 1 again, and eventually encounter a 'new' 1.0.1

Reply:
If I understand what you mean, I don't tinnk it is a problem, usually I saw people wiring < 1.0 before an official release (I guess the paper?) and then it updates progressively. So the paper is perfectly fine, when you submit it is sufficient to write python setup_version --bump_major, and commit the changes (it will become immuneMl 1.0.0). It can be also 1.0.1 with no problem is that is the preference. A pipy package can be then added with the consistent label-release.

Reply:
Enrico, we are also using standard labeling. Instead of having a command that changes the version we are changing the content of the file. It is important for the paper to keep the exact labeling as we have now so that the results can be replicated exactly (some options differ between versions). I agree that having an automated version would be beneficial, just not sure if this is the way to go at this moment?

Reply:
it can also be set by hand now by changing the version.json directly. I set it to 1.0.1 release now, is it right?